### ISSUE TITLE
Feature: CSV playlists should switch to relative paths (but allow older abspath format)

### ISSUE THREAD

**Issue by mpogue2** [mpogue2/SquareDesk#174]
Right now, the header of the CSV file is "abspath", and the paths are absolute.
This is fragile, if the music directory is renamed or moved (it will break all existing playlists).

To provide for backward compatibility, it would be nice to do this:
- change "abspath" to "relpath" (this essentially marks the V2 of the playlist .csv format)
- write paths relative to the root of the Music Directory for CSV files
- change reading to look relative to the root of the Music Directory
- retain backwards compatibility by still allowing reads of "abspath"-style playlists

I believe that M3U files need to still be absolute, because that's the only way other programs will be able to use the (standard format) M3U file as a playlist.
- Change the default to use CSV-style playlists.

**Comment by mpogue2** [comment]
Changing to a Bug.  As it stands, playlists are incompatible between Mac OS X and Win32.  So, if you're using Dropbox to share everything (incl playlists), they can only be read properly by the OS that wrote them.   Switching to relative pathnames will fix this.

**Comment by mpogue2** [comment]
Fixed, with some backward compatibility with existing absolute playlists.
If the first line of a playlist contains "abspath", then it's a V1 playlist, using absolute pathnames.
OTOH, if it contains "relpath", then it's a V2 playlist, using relative pathnames.
The effects of this change are:
- existing absolute V1 playlists continue to work as they do today, under normal operation.  
- existing V1 playlists do not work across platforms shared with Dropbox or Google Drive
- existing V1 playlists stop working, if you move the musicDir.
- the new, relative V2 playlists will work across platforms shared with Dropbox or Google Drive
- the new V2 playlists will work, if the musicDir is moved.  Just move the playlists along, too.
- and finally, all new saved playlists will be saved in the V2 relative format

If you have old V1 playlists, they can be converted to V2 playlists by manually editing, or by a sed command, for example:

old V1 format:
```
abspath,pitch,tempo
"/Users/mpogue/Google Drive/__squareDanceMusic/patter/ST 1031 - Love Boat (intro).mp3",-1,128

```
new format (initial "/" now means musicRootPath):
```
relpath,pitch,tempo
"/patter/ST 1031 - Love Boat (intro).mp3",-1,128
```

I am not providing an automated conversion utility from V1 to V2 at this time.  I'm just going to "sed" my playlists, with a custom 1-line sed command.

**Comment by mpogue2** [comment]
Fixed by dcfcd7bca3ca17d6ec3d6a9a156d74b78d0daf7b .
Wow.  2 1/2 years to fix this one!  I finally decided to fix it, when I migrated from Dropbox to Google Drive, and all the existing playlists broke.
